### PR TITLE
honor use_proxy parameter

### LINKED ISSUE #58632: use_proxy is documented but never used in uri module
##### SUMMARY
`use_proxy` is described as module parameter but is never used in the code. 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
`uri` module

##### ANSIBLE VERSION
```
ansible 2.8.1
```

##### CONFIGURATION
```
-
```

##### OS / ENVIRONMENT
all

##### STEPS TO REPRODUCE

We use a corporate proxy, except for the desired host. So i put `use_proxy: no` into the task like this:
```
     - uri:
        url: "https://<example url>/"
        use_proxy: no
```

The result was the following error in output:
```
"msg": "Status code was -1 and not [200]: Request failed: <urlopen error Tunnel connection failed: 404 Not Found>"
```

I inspected the code:
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/net_tools/basics/uri.py

At line 160, the parameter is described. But this is the only place in this file. So i think, the parameter can not work.

##### EXPECTED RESULTS
`use_proxy: no` should work as intended like in `get_url`

##### ACTUAL RESULTS
`use_proxy: no` does not work.

### PR DESCRIPTION
**Pull request by h4rr21**
##### SUMMARY
Fixes #58632

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
the same as
https://github.com/ansible/ansible/pull/65706
with a rebase

### CODE REVIEW

**Comment by bcoca** [comment]
iirc, this is not needed, it is actually pulled from the first parameter 'module'

**Comment by h4rr21** [comment]
hi @bcoca , any idea why CI is failing with 
```
01:00 ERROR: Found 1 changelog issue(s) which need to be resolved:
01:00 ERROR: changelogs/fragments/58632-uri-include_use_proxy.yaml :0:0: extension must be one of: .yml, .yaml (0%)
```
extension is `yaml`

**Comment by s-hertel** [comment]
It looks like the extension has a trailing space:
```
01:14 ERROR: Found 1 no-illegal-filenames issue(s) which need to be resolved:
01:14 ERROR: changelogs/fragments/58632-uri-include_use_proxy.yaml :0:0: illegal file name end-char ' ' (0%)
```

```
$ ls changelogs/fragments/58632-uri-include_use_proxy.yaml\ 
'changelogs/fragments/58632-uri-include_use_proxy.yaml '
```

**Comment by h4rr21** [comment]
fixed, thanks @s-hertel 👍 

> It looks like the extension has a trailing space:
> 
> ```
> 01:14 ERROR: Found 1 no-illegal-filenames issue(s) which need to be resolved:
> 01:14 ERROR: changelogs/fragments/58632-uri-include_use_proxy.yaml :0:0: illegal file name end-char ' ' (0%)
> ```
> 
> ```
> $ ls changelogs/fragments/58632-uri-include_use_proxy.yaml\ 
> 'changelogs/fragments/58632-uri-include_use_proxy.yaml '
> ```

**Comment by h4rr21** [comment]
is there any proxy server in the Ansible CI available for testing ?

**Comment by bcoca** [comment]
reading the other issues, they seem to be stalled because core team memeber wanted a more 'complete' solution, something like this:

```
diff --git a/lib/ansible/module_utils/urls.py b/lib/ansible/module_utils/urls.py
index f34c536890..5cfc42ebbc 100644
--- a/lib/ansible/module_utils/urls.py
+++ b/lib/ansible/module_utils/urls.py
@@ -1727,7 +1727,7 @@ def url_argument_spec():


 def fetch_url(module, url, data=None, headers=None, method=None,
-              use_proxy=True, force=False, last_mod_time=None, timeout=10,
+              use_proxy=None, force=False, last_mod_time=None, timeout=10,
               use_gssapi=False, unix_socket=None, ca_path=None, cookies=None, unredirected_headers=None):
     """Sends a request via HTTP(S) or FTP (needs the module as parameter)

@@ -1776,6 +1776,9 @@ def fetch_url(module, url, data=None, headers=None, method=None,
     # Get validate_certs from the module params
     validate_certs = module.params.get('validate_certs', True)

+    if use_proxy is None:
+        use_proxy = module.params.get('use_proxy', True)
+
     username = module.params.get('url_username', '')
     password = module.params.get('url_password', '')
     http_agent = module.params.get('http_agent', 'ansible-httpget')

```

That in ADDITION to the existing code in the PR would solve the issue for any other modules that had the same issue with 'forgetting' to add the parameter.

**Review by bcoca — APPROVED** [review]

File: lib/ansible/module_utils/urls.py [hunk 1]
```diff
@@ -1727,7 +1727,7 @@ def url_argument_spec():
 
 
 def fetch_url(module, url, data=None, headers=None, method=None,
-              use_proxy=True, force=False, last_mod_time=None, timeout=10,
+              use_proxy=None, force=False, last_mod_time=None, timeout=10,
```
Comment (sivel) [inline lib/ansible/module_utils/urls.py:1730]: The fact that `use_proxy` is an explicit kwarg, means that `module` is not the right place to get the arg from.

What needs to happen, is that the `uri` module needs to explicitly pass `module.params['use_proxy']` to `fetch_url` instead.

We've effectively stopped using the ability to just use `module.params` directly here, and have instead been making them explicit kwargs.

**Review by sivel — CHANGES_REQUESTED** [review]

File: lib/ansible/module_utils/urls.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/module_utils/urls.py:1730]: uri module still passes this explicitly, this was added due to my comment above (which is really reflecting the reason this issue stalled in previous incarnations). The change really only covers 'other modules' that include the option but don't pass it in.

Comment (h4rr21) [reply lib/ansible/module_utils/urls.py:1730]: yep, as mentioned in the last comment 

> What needs to happen, is that the uri module needs to explicitly pass module.params['use_proxy'] to fetch_url instead.

this change it's included in my PR. @sivel 

https://github.com/ansible/ansible/blob/dcd9fe8bccfd42698792d79a135f2d0a231d51b2/lib/ansible/modules/uri.py#L595

**Review by sivel — APPROVED** [review]

**Pull request merged**